### ISSUE TITLE
Unable to use created resource pool and vm created in the same step

### ISSUE THREAD

**Issue by Russell-IO** [terraform-providers/terraform-provider-vsphere#593]
### Terraform Version
Terraform v0.11.7

### vSphere Provider Version
* provider.vsphere: version = "~> 1.6"

### Affected Resource(s)

Please list the resources as a list, for example:
- `vsphere_virtual_machine`
- `vsphere_resource_pool`

### Terraform Configuration Files

```hcl
variable "datacentre" {}
variable "cluster" {}
variable "template" {}
variable "count" { default = 1 }
variable "virtual_machine_name_prefix" { default = "test" }
variable "network" { default = "VM Network" }
variable "virtual_machine_domain" { default = "example.lab"}
variable "virtual_machine_network_address" { default = "192.168.0.0/24" }
variable "virtual_machine_ip_address_start" { default = "250"}
variable "virtual_machine_cpu" { default = 2 }
variable "virtual_machine_memory" { default = 2048}
variable "virtual_machine_network_gateway" { default = "192.168.0.1" }

data "vsphere_datacenter" "main" {
  name = "${var.datacentre}"
}

data "vsphere_datastore" "vsan" {
  name = "VSAN"
  datacenter_id = "${data.vsphere_datacenter.main.id}"
}

data "vsphere_resource_pool" "main" {
  name = "${var.cluster}/Resources/"
  datacenter_id = "${data.vsphere_datacenter.main.id}"
}

resource "vsphere_resource_pool" "main" {
  name = "test-rp"
  parent_resource_pool_id = "${data.vsphere_resource_pool.main.id}"
}

data "vsphere_compute_cluster" "main" {
  name = "${var.cluster}"
  datacenter_id = "${data.vsphere_datacenter.main.id}"
}

data "vsphere_virtual_machine" "template" {
  name = "${var.template}"
  datacenter_id = "${data.vsphere_datacenter.main.id}"
}

data "vsphere_network" "main" {
  name          = "${var.network}"
  datacenter_id = "${data.vsphere_datacenter.main.id}"
}

resource "vsphere_virtual_machine" "main" {
  count            = "${var.count}"
  name             = "${var.virtual_machine_name_prefix}${count.index + 1}"
  resource_pool_id = "${vsphere_resource_pool.main.id}"
  datastore_id     = "${data.vsphere_datastore.vsan.id}"

  num_cpus = "${var.virtual_machine_cpu}"
  memory   = "${var.virtual_machine_memory}"
  guest_id = "${data.vsphere_virtual_machine.template.guest_id}"

  wait_for_guest_net_routable = false

  network_interface {
    network_id   = "${data.vsphere_network.main.id}"
    adapter_type = "${data.vsphere_virtual_machine.template.network_interface_types[0]}"
  }

  disk {
    label = "disk0"
    size  = "${data.vsphere_virtual_machine.template.disks.0.size}"
  }

  clone {
    template_uuid = "${data.vsphere_virtual_machine.template.id}"

    customize {
      linux_options {
        host_name = "${var.virtual_machine_name_prefix}${count.index + 1}"
        domain    = "${var.virtual_machine_domain}"
      }

      network_interface {
        ipv4_address = "${cidrhost(var.virtual_machine_network_address, var.virtual_machine_ip_address_start + count.index)}"
        ipv4_netmask = 24
      }

      ipv4_gateway = "${var.virtual_machine_network_gateway}"

    }
  }

}
```

### Expected Behavior

Terraform should create the resource group first, then create the VM

### Actual Behavior

⇒  terraform plan
Refreshing Terraform state in-memory prior to plan...
The refreshed state will be used to calculate this plan, but will not be
persisted to local or remote state storage.

data.vsphere_datacenter.main: Refreshing state...
data.vsphere_virtual_machine.template: Refreshing state...
data.vsphere_datastore.vsan: Refreshing state...
data.vsphere_network.main: Refreshing state...
data.vsphere_resource_pool.main: Refreshing state...
data.vsphere_compute_cluster.main: Refreshing state...

------------------------------------------------------------------------

Error: Error running plan: 1 error(s) occurred:

* vsphere_virtual_machine.main: 1 error(s) occurred:

* vsphere_virtual_machine.main: could not find resource pool ID "": ServerFaultCode: The object 'vim.ResourcePool:' has already been deleted or has not been completely created

### Steps to Reproduce

Please list the steps required to reproduce the issue, for example:
1. `terraform apply`

### Important Factoids

If I hash out the virtual machine resource and execute the steps to create the resource pool first, then unhash the virtual machine resource, it works.

**Comment by Russell-IO** [comment]
@bill-rich - what's occurring on your side ? it seems to happen when I try creating a resource pool & vm in the same `terraform apply`, please see below, n.b. if I hash out the vm, create the resource pool first, then unhash out the VM it works.

For example with the AWS provider, if you create a launch configuration and autoscaling group in one step the autoscaling group accepts a `computed` state, but this provider doesn't seem to

https://gist.github.com/Russell-IO/f648ed889eec29fb11afcbdf67b2c162

**Comment by jacob-carlborg** [comment]
I have the same problem creating a datacenter and a computer cluster:

```hcl
provider "vsphere" {
  allow_unverified_ssl = true
  version = "1.6.0"
}

locals {
  datacenter = "SST"
  compute_cluster = "sst-cluster"
}

resource "vsphere_datacenter" "datacenter" {
  name = "${local.datacenter}"
}

resource "vsphere_compute_cluster" "compute_cluster" {
  name = "${local.compute_cluster}"
  datacenter_id = "${vsphere_datacenter.datacenter.id}"

  drs_enabled = true
  drs_automation_level = "fullyAutomated"

  ha_enabled = true
}
```

Trace output: https://gist.github.com/jacob-carlborg/d0ab1be997f8e2b7cde45e430c95d8e4.

**Comment by bill-rich** [comment]
Thanks for the additional info @Russell-IO and @jacob-carlborg! I am seeing the dependencies being correctly calculated when I'm testing, but there certainly seems to be something here. I'm going to take a closer look at the attached logs and see what I was missing on my first attempts.

**Comment by jacob-carlborg** [comment]
I can add that I’m trying to set up an new datacenter from scratch.

**Comment by Russell-IO** [comment]
@skywalk7 thanks for that, it looks like 1.7.0 has been released which looks like it addresses this, I will test and confirm:

BUG FIXES:

resource/vsphere_virtual_machine: The resource no longer attempts to set ResourceAllocation on virtual ethernet cards when the vSphere version is under 6.0. (#579)
resource/vsphere_resource_pool: The read function is now called at the end of resource creation. (#560)
Updated govmomi to v0.18. (#600)

**Comment by bill-rich** [comment]
@jacob-carlborg Looking more closely at your problem, it was fixed in #575. You'll need to used `moid` for the datacenter instead of `id`. The datacenter resource is an older one and is using the name instead of the Managed Object ID for the `id`. To not break backward compatibility the `moid` attribute was added.

**Comment by bill-rich** [comment]
@Russell-IO #560 actually won't cover this one, but I am putting together a fix that will. It will be available in the next release.

I'm going to close this and track the issue in #533.
